website/sales/models/order.py

```diff
@@ -129,18 +129,33 @@ class Meta:
         Coalesce(Sum("order_items__amount"), Value(0), output_field=IntegerField())
     )
 
+    _is_free = models.BooleanField(
+        verbose_name=_("is free"),
+        default=False,
+    )
+    _total_amount = PaymentAmountField(
```

Comment:
To be sure that the fields stay up to date, I guess the stored field should also be updated in a post_save signal from OrderItems.
With that done, it should also be fine to remove the `total_amount` annotation property, so there is a single source of truth, efficiency everywhere, and no double code to maintain.

Reply:
Oh actually, I see you already do that from `OrderITem.save`, that works. Still then I think it would be preferable to have a single source of truth.

Reply:
We already do a order.save() in the order item save method. 

Sadly, the post_save signal does not give guarantees. For example, deletions don't always call post_save signals. The choice to make `_total_amount` be a cached, last-computed value that _might_ be outdated was deliberate. Querying total_amount however will always return the guaranteed correct value

Reply:
Well it's true that there's no strict guarantees this way, but as long as we don't do bulk updates or bulk deletes there are guarantees. And we shouldn't be doing bulk operations anyway (that would also break 'immutable' payables for example).

Reply:
Then still, the current aggregate property remains the most efficient way to actually calculate _total_amount so we can't remove it (or I would remove it and replace it by less efficient Python code)

Reply:
@DeD1rk what do you think?

Reply:
Fair enough, there's one strange thing though: the save code uses an in-python calculation instead of just getting the current value of the total amount property

Reply:
I see, probably relates to the comment I removed `# We cannot use self.total_amount as it is a requires a database query and hence will not use any updated values`. I will test it